qtglean/src/ffi/ping.rs

```diff
@@ -2,9 +2,11 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this
 // file, You can obtain one at https://mozilla.org/MPL/2.0/.
 
+use ffi_support::FfiStr;
 use crate::metrics::__generated_pings::submit_ping_by_id;
 
+
 #[no_mangle]
-pub extern "C" fn glean_submit_ping_by_id(id: u32) {
-    submit_ping_by_id(id, None);
+pub extern "C" fn glean_submit_ping_by_id(id: u32, reason: FfiStr) {
+    submit_ping_by_id(id, Some(&reason.into_string()));
```

Comment:
This is going to log an error `Error: Invalid reason code <reasonCode> for ping <ping>` any time an invalid or no reason code is sent. Not sure how to work around this since FfiStr doesn't appear to allow it's value to be copied. 

@brizental @mcleinman any suggestions?

Reply:
Getting an error for an invalid reason code seems fair, no?

For no reason, though: Maybe something similar to what's in `timing_distribution.rs`:
```
let value_as_json = match value {
        Some(v) => serde_json::to_string(&v).unwrap(),
        None => "".to_string(),
    };
```

Reply:
For invalid this is fine. For no reason code no. If no reason code is provided, you should give `submit_ping_by_id` a `None` value, instead of a `Some`.